src/record_merge.jl

```diff
@@ -0,0 +1,12 @@
+"""
+    record_merge(record; fields_to_merge...)
+
+Return a new `AbstractRecord` with the same schema as `record` by merging the fields of
+`record` and `fields_to_merge` together using `Tables.rowmerge`.
+"""
```

Comment:
```suggestion
"""
    record_merge(record::AbstractRecord; fields_to_merge...)

Return a new `AbstractRecord` with the same schema version as `record`, whose fields
are computed via `Tables.rowmerge(record; fields_to_merge...)`. The returned record is 
constructed by passing these merged fields to the `AbstractRecord` constructor matching 
the type of the input `record`.
"""
```

just some additional clarity for callers